org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/commons/ParentProcessWatcher.java

```diff
@@ -83,14 +83,32 @@ private boolean parentProcessStillRunning() {
 		} else {
 			command = "ps -p " + pid;
 		}
+		Process process = null;
+		boolean finished = false;
 		try {
-			Process process = Runtime.getRuntime().exec(command);
-			int processResult = process.waitFor();
-			return processResult == 0;
+			process = Runtime.getRuntime().exec(command);
+			finished = process.waitFor(POLL_DELAY_SECS, TimeUnit.SECONDS);
+			if (!finished) {
+				process.destroy();
+				finished = process.waitFor(POLL_DELAY_SECS, TimeUnit.SECONDS); // wait for the process to stop
+			}
+			return !finished || process.exitValue() == 0;
 		} catch (IOException | InterruptedException e) {
-			//JavaLanguageServerPlugin.logException(e.getMessage(), e);
+			// JavaLanguageServerPlugin.logException(e.getMessage(), e);
 			return true;
+		} finally {
+			if (process != null) {
+				if (!finished) {
+					process.destroyForcibly();
+				}
+				// Java doesn't call the finalize method when terminating a process. 
+				// Because of that, when Java LS isn't used, it creates a lot of zombie processes on Windows.
+				// GC calls finalize
+				// We can also call it on Linux and Mac because it runs when Java LS isn't active.

```

Comment:
While calling the GC on an idle server is not a big deal on Linux and Mac, it will still cause unnecessary CPU usage at this point. So we might as well do it only for Windows

Reply:
I have updated the PR.